src/couch_index/src/couch_index_server.erl

```diff
@@ -187,18 +285,20 @@ code_change(_OldVsn, State, _Extra) ->
     {ok, State}.
 
 
-handle_config_change("couchdb", "index_dir", RootDir, _, RootDir) ->
-    {ok, RootDir};
-handle_config_change("couchdb", "view_index_dir", RootDir, _, RootDir) ->
-    {ok, RootDir};
+handle_config_change("couchdb", "index_dir", RootDir, _, #st{root_dir=RootDir}=State) ->
```

Comment:
It is not quite right. The state here is the value we pass to [`config:listen_for_changes/2`](https://github.com/apache/couchdb/pull/477/files#diff-3f723e40663b52d70b47cbd302216517R193)